opengrok-indexer/src/test/java/org/opengrok/indexer/web/StatisticsTest.java

```diff
@@ -240,34 +251,68 @@ public void testFrom() {
+        while ((token = jParser.nextToken()) != JsonToken.END_OBJECT) {
+            if (token == JsonToken.START_OBJECT) {
+                while (jParser.nextToken() != JsonToken.END_OBJECT) {
+                }
+            }
+
+            if (token != JsonToken.FIELD_NAME)
+                continue;
+
+            String fieldname = jParser.getCurrentName();
+            if (name.equals(fieldname)) {
+                jParser.nextToken();
+                value = jParser.getLongValue();
+                break;
+            }
+        }
+        jParser.close();
+
+        return value;
+    }
+
+    protected void checkToJson(Function<Statistics, String> callback) throws IOException {
         Statistics stats = new Statistics();
         stats.setRequests(145L);
         stats.setMinutes(-325L);
         stats.setRequestsPerMinute(1000L);
         stats.setRequestsPerMinuteMin(107L);
         stats.setRequestsPerMinuteMax(106L);
-        JSONObject result = callback.apply(stats);
-
+        String result = callback.apply(stats);
         Assert.assertNotNull(result);
-        Assert.assertNotNull(result.get(Statistics.STATISTIC_MINUTES));
-        Assert.assertEquals(-325L, (long) result.get(Statistics.STATISTIC_MINUTES));
-        Assert.assertNotNull(result.get(Statistics.STATISTIC_REQUESTS));
-        Assert.assertEquals(145L, (long) result.get(Statistics.STATISTIC_REQUESTS));
-        Assert.assertNotNull(result.get(Statistics.STATISTIC_REQUESTS_PER_MINUTE));
-        Assert.assertEquals(1000L, (long) result.get(Statistics.STATISTIC_REQUESTS_PER_MINUTE));
-        Assert.assertNotNull(result.get(Statistics.STATISTIC_REQUESTS_PER_MINUTE_MIN));
-        Assert.assertEquals(107L, (long) result.get(Statistics.STATISTIC_REQUESTS_PER_MINUTE_MIN));
-        Assert.assertNotNull(result.get(Statistics.STATISTIC_REQUESTS_PER_MINUTE_MAX));
-        Assert.assertEquals(106L, (long) result.get(Statistics.STATISTIC_REQUESTS_PER_MINUTE_MAX));
+
+        Assert.assertNotNull(getJsonLongField(result, Statistics.STATISTIC_MINUTES));
+        Assert.assertEquals(-325L, (long) getJsonLongField(result, Statistics.STATISTIC_MINUTES));
+        Assert.assertNotNull(getJsonLongField(result, Statistics.STATISTIC_REQUESTS));
+        Assert.assertEquals(145L, (long) getJsonLongField(result, Statistics.STATISTIC_REQUESTS));
+        Assert.assertNotNull(getJsonLongField(result, Statistics.STATISTIC_REQUESTS_PER_MINUTE));
+        Assert.assertEquals(1000L, (long) getJsonLongField(result, Statistics.STATISTIC_REQUESTS_PER_MINUTE));
+        Assert.assertNotNull(getJsonLongField(result, Statistics.STATISTIC_REQUESTS_PER_MINUTE_MIN));
+        Assert.assertEquals(107L, (long) getJsonLongField(result, Statistics.STATISTIC_REQUESTS_PER_MINUTE_MIN));
+        Assert.assertNotNull(getJsonLongField(result, Statistics.STATISTIC_REQUESTS_PER_MINUTE_MAX));
+        Assert.assertEquals(106L, (long) getJsonLongField(result, Statistics.STATISTIC_REQUESTS_PER_MINUTE_MAX));
```

Comment:
is there any benefit in doing this complex stuff with getjsonLongField instead of using Statistics.fromJson and then comparing the values with expected values.

Reply:
This sort of follows the idea done previously. I realized it can serve also as a test that will make sure the JSON serialization has a certain form.

Reply:
I concur, this is more about structure, not whitespaces. That's why I think using generator is better than the test in `StatisticsUtilTest#testLoadStatistics()`.

Reply:
On the other hand, using concrete JSON representation is closer to whitebox testing - e.g. testing idempotence can pass without making sure that any JSON processing actually happened which is arguably not a problem when calling a generator from JSON library :-)